### PR TITLE
Project setup

### LINKED ISSUE #5: Set up exporting of backend functions api to renderer process
We will need to export all of the 'backend' functions to the renderer process, so we can call them from the GUI. Create appropriate folders to define the backend functions, and create index.js file to exports all functions. One would need to import this in the preload.js and export in the preload. Make sure to read the docs of template to know the security aspects of preload.js nad requirements.

### PR DESCRIPTION
**Pull request by ADIthaker**
### Description

> 1. `mainEvents` contains all files that handle ipc calls for main process and the files are imported into the `main.js` file.
> 2. The files for ipcMain events **should** be inside **folders(named by functionality)** inside `mainEvents`.
> 3. `renderEvents` contains all files that handle ipc calls for renderer process and the those files are imported into `renderEvents/index.js` and index.js is then imported into `preload.js` where the renderer process communication can be set on the window object using `contextBridge`.
> 4. The files for `ipcRenderer` events can be inside separate folders inside `renderEvents` and imported appropriately in the exports folder in `index.js` file
> 5. Added `utils` folder for any utility functions or objects.
> 6. `loadMainProcess` function loads all files that contain `ipcMain` events from `mainEvents` function
> 

### Changes

> - Added folder `mainEvents` in `app/electron/` and added `.gitkeep` file in it
> - Added folder `renderEvents` in `app/electron/` and added `index.js` file in it
> - Imported `renderEvents/index.js` in `preload.js` in `app/electron/`
> - Added `loadMainProcess` function in `main.js` file to import all `ipcMain` events from `mainEvents` folder.

### Issues
   Resolves #5 

### Impacted Area

- Project Structure

### Steps to Test

-  `loadAndProcess` function

### CODE REVIEW

File: app/electron/main.js [hunk 1]
```diff
@@ -41,6 +42,13 @@ async function createWindow() {
     // so I don't have to write another 'if' statement
     protocol.registerBufferProtocol(Protocol.scheme, Protocol.requestHandler);
   }
+  const loadMainProcess = () => {
+    const files = glob.sync(path.join(__dirname, "mainEvents/**/*.js"));
+    /* eslint-disable global-require, import/no-dynamic-require */
```
Comment (YJDoc2) [inline app/electron/main.js:47]: do not disable eslint, let there be warnings

File: app/electron/main.js [hunk 2]
```diff
@@ -114,7 +122,8 @@ async function createWindow() {
     // before the DOM is ready
     win.webContents.once("dom-ready", () => {
       // win.webContents.openDevTools(); // see https://github.com/reZach/secure-electron-template/issues/48
-      require("electron-debug")(); // https://github.com/sindresorhus/electron-debug
+      require("electron-debug")(); // eslint-disable-line global-require
```
Comment (YJDoc2) [inline app/electron/main.js:125]: do not disable eslint, let there be warnings

File: app/electron/menu.js [hunk 3]
```diff
@@ -195,7 +195,7 @@ const MenuBuilder = (mainWindow, appName) => {
           {
             label: "Learn More",
             click: async () => {
-              const { shell } = require("electron");
+              const { shell } = require("electron"); // eslint-disable-line global-require
```
Comment (YJDoc2) [inline app/electron/menu.js:198]: do not disable eslint

**Review by YJDoc2 — APPROVED** [review]
Just do not disable eslint, if there are warnings, let there be warnings

**Review by namandangi — APPROVED** [review]
Don't disable eslint warnings

File: app/electron/main.js (hunk 1 above)
Comment (ADIthaker) [reply app/electron/main.js:47]: It doesn't let me commit the changes if there are warnings

**Pull request merged**